config/xp_role.go

```diff
@@ -17,181 +23,446 @@ type XpRole struct {
+		Components: []component.Message{
+			&component.TextDisplay{Content: "## Configuration / Rôles de niveaux"},
+			&component.Separator{},
+			&component.Section{
+				Components: []component.Message{
+					&component.TextDisplay{Content: fmt.Sprintf("Niveau **%d**", exp.Level(role.XP))},
+				},
+				Accessory: &component.Button{
+					CustomID: fmt.Sprintf("xp_role_edit_level_start_%d", id),
+					Style:    component.ButtonStyleSecondary,
+					Label:    "Modifier",
+				},
+			},
+			&component.ActionsRow{Components: []component.Message{roleSelect}},
+			&component.ActionsRow{Components: []component.Message{
+				&component.Button{CustomID: fmt.Sprintf("xp_role_del_%d", id), Style: component.ButtonStyleDanger, Label: "Supprimer"},
+			}},
+			&component.Separator{},
+			&component.ActionsRow{Components: []component.Message{
+				&component.Button{Label: "Retour", CustomID: ModifyXpRole, Style: component.ButtonStyleSecondary},
+			}},
+		},
+	}
+
+	response := &interaction.Response{
+		Type: types.InteractionResponseUpdateMessage,
+		Data: &interaction.ResponseData{
+			Components: []component.Component{container},
+			Flags:      channel.MessageFlagsIsComponentsV2,
+		},
+	}
+
+	err = session.InteractionAPI().Respond(i.Interaction, response)
+	if err != nil {
+		session.LogError(err, "Sending xp_role config")
+	}
 }
 
-func HandleXpRoleAddRole(s *discordgo.Session, i *discordgo.InteractionCreate, data interaction.MessageComponentData, resp *cmd.ResponseBuilder) {
-	resp.IsEphemeral()
+func HandleXpRoleEditRole(
+	session *discordgo.Session,
+	i *discordgo.InteractionCreate,
+	data interaction.MessageComponentData,
+	parameters []string, resp *cmd.ResponseBuilder,
+) {
+	id, err := getRoleLevelID(parameters)
+	if err != nil {
+		session.LogError(err, "Reading dynamic CustomID")
+		return
+	}
+	role := data.Values[0]
 	cfg := GetGuildConfig(i.GuildID)
-	roleId := data.Values[0]
-	for _, r := range cfg.XpRoles {
-		if r.RoleID == roleId {
-			err := resp.SetMessage("Le rôle est déjà présent dans la config").Send()
+	for _, xpRole := range cfg.XpRoles {
+		if xpRole.RoleID == role {
+			err = session.InteractionAPI().Respond(i.Interaction, &interaction.Response{
+				Type: types.InteractionResponseChannelMessageWithSource,
+				Data: &interaction.ResponseData{
+					Flags:           channel.MessageFlagsEphemeral,
+					AllowedMentions: &channel.MessageAllowedMentions{},
+					Content:         fmt.Sprintf("Un autre niveau avec le rôle <@&%s> est déjà existant.", role),
+				},
+			})
 			if err != nil {
-				s.LogError(err, "sending role already in config")
+				session.LogError(err, "Sending unable to Already existing role message")
 			}
 			return
 		}
 	}
-	cfg.XpRoles = append(cfg.XpRoles, XpRole{
-		XP:     configModifyMap[getKeyConfigRole(i)],
-		RoleID: roleId,
-	})
-	err := cfg.Save()
-	if err != nil {
-		s.LogError(err, "saving config for role %s in %s", roleId, i.GuildID)
+	index, xprole := cfg.FindXpRoleID(id)
+	if index == 0 {
```

Comment:
A nil check is better here because we don't need the index:
```go
_, xprole := cfg.FindXpRoleID(id)
if xprole == nil {
```
and `0` is a valid return value for `Guild.FindXpRoleID`